pkg/operator/cluster/ceph/osd/osd.go

```diff
@@ -131,26 +152,28 @@ func (c *Cluster) Start() error {
 	}()
 
 	if c.Storage.UseAllNodes {
-		// make a daemonset for all nodes in the cluster
-		ds := c.makeDaemonSet(c.Storage.Selection, c.Storage.Config)
-		_, err := c.context.Clientset.Extensions().DaemonSets(c.Namespace).Create(ds)
+		// resolve all storage nodes
+		c.Storage.Nodes = nil
+		rookSystemNS := os.Getenv(k8sutil.PodNamespaceEnvVar)
```

Comment:
yes, osd pod has rbac permission to access configmaps in cluster namespace. 

here because `rookSystemNS` is also used as the cluster name, so it is extracted here and passed to discovery Pod as cluster name, rather than accessing objects in rook system namespace.